src/components/App.jsx

```diff
@@ -28,6 +30,18 @@ function App() {
   const [currentKeyword, setCurrentKeyword] = useState('');
 
   const location = useLocation();
+  
+  useEffect(() => {
+    const token = api.getToken();
+    if (token) {
+      api.getUser()
+        .then((user) => {
+          setCurrentUser(user);
+          setIsLoggedIn(true);
+        })
+        .catch(() => api.logout()); 
```

Comment:
`        .catch(() => api.logout()); 
`

You don't need to call `api.logout(` if the autologin failed.